pkg/reconciler/taskrun/resources/apply.go

```diff
@@ -97,6 +98,15 @@ func ApplyResources(spec *v1beta1.TaskSpec, resolvedResources map[string]v1beta1
 	return ApplyReplacements(spec, replacements, map[string][]string{})
 }
 
+// ApplyPipelineTaskContexts applies the substitution from $(context.pipelineTask.*) with the specified values.
+// Uses "0" as a default if a value is not available.
+func ApplyPipelineTaskContexts(spec *v1beta1.TaskSpec, pt *v1beta1.PipelineTask) *v1beta1.TaskSpec {
+	replacements := map[string]string{
+		"context.pipelineTask.retries": strconv.Itoa(pt.Retries),
```

Comment:
This variable is only limited to a `pipelineTask` with an inlined `taskSpec`. Is that by design? `pipelineTask` does imply that the task has to be `pipelineTask` but `pipelineTask` can either have `taskRef` or `taskSpec` and `retries` are supported with both possibilities.

Reply:
my assumption these would be available to use as param values - and using them in an inline spec feels like a bonus

(i dont think Tasks should be able to refer to these values since they need to work outside of the context of pipelines)

Reply:
I'm unsure if `pipelineTask` can be specified with `taskRef` either. It's a little bit weird that a variable called `context.pipelineTask.*` in the spec of an individual `Task`.

Reply:
please see my [comment](https://github.com/tektoncd/pipeline/pull/3770/files#r601923749) 🆙